MQTTClient-C/src/linux/MQTTLinux.h

```diff
@@ -46,29 +46,17 @@
 #include <string.h>
 #include <signal.h>
 
+#include "Network.h"
+#include "Timer.h"
+
 typedef struct Timer
 {
 	struct timeval end_time;
 } Timer;
 
-void TimerInit(Timer*);
-char TimerIsExpired(Timer*);
-void TimerCountdownMS(Timer*, unsigned int);
-void TimerCountdown(Timer*, unsigned int);
-int TimerLeftMS(Timer*);
-
 typedef struct Network
 {
 	int my_socket;
-	int (*mqttread) (struct Network*, unsigned char*, int, int);
```

Comment:
As an alternative approach, you can keep the function pointers in the `Network` interface and implement them for each platform. Then, each platform provides the `NetworkInit()` function, which initializes these pointers to platform-specific functions. This way, you define the `Network` structure once in the common header and don't need any forward declarations.